src/test/java/liuni/LiUniResourceTest.java

```diff
@@ -84,30 +84,32 @@ public void testREST_fetchTimeline_emptyTimeline() {
 
     @Test
     public void testREST_fetchTimeline_updateTimeline() {
-        ResponseList<Status> respList = new ResponseListImpl<Status>();
+        List<TwitterTweetModel> tweetModelList = new ArrayList<TwitterTweetModel>();
         try {
-            doReturn(respList).when(twitter).getHomeTimeline();
+            TwitterService twitterService = mock(TwitterService.class);
+            when(twitterService.getTimeline()).thenReturn(tweetModelList);
+            resource.setTwitterService(twitterService);
 
             Response resp = resource.fetchTimeline();
 
             assertEquals(resp.getStatus(), Response.Status.OK.getStatusCode());
-            assertEquals(resp.getEntity(), respList);
+            assertEquals(tweetModelList, resp.getEntity());
 
-            respList.add(status);
+            tweetModelList.add(mock(TwitterTweetModel.class));
             resp = resource.fetchTimeline();
 
             assertEquals(Response.Status.OK.getStatusCode(), resp.getStatus());
-            assertEquals(respList, resp.getEntity());
+            assertEquals(tweetModelList, resp.getEntity());
         }
         catch (Exception e) {
             Assert.fail("This exception is not expected.");
         }
     }
 
     @Test
-    public void testREST_fetchTimeline_fetchTimelineError() {
-        ResponseList<Status> respList = new ResponseListImpl<Status>();
-        respList.add(status);
+    public void testREST_fetchTimeline_timeline() {
```

Comment:
make the name better for what is happening

Reply:
Done